### PR TITLE
remove nameRegex from validators

### PR DESCRIPTION
**Pull request by MariyaSin**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/783?issue_count=12&issue_position=3&next_issue_id=782&prev_issue_id=784

### CODE REVIEW

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 1]
```diff
@@ -101,8 +99,7 @@ private void HandleInternalPropertyValidation(Operation<EditUserRequest> request
                 new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
                 {
                     { x => !string.IsNullOrEmpty(x.value.ToString()), "FirstName is too short" },
```
Comment (TIove) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:101]: “FirstName is empty”

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 2]
```diff
@@ -12,30 +12,26 @@ public CreateUserRequestValidator()
             RuleFor(user => user.FirstName)
                 .NotEmpty()
                 .MaximumLength(32).WithMessage("First name is too long.")
-                .MinimumLength(1).WithMessage("First name is too short.")
-                .Matches("^[A-Z][a-z]+$|^[А-ЯЁ][а-яё]+$").WithMessage("First name with error.");
+                .MinimumLength(1).WithMessage("First name is too short.");
```
Comment (BelehovEgor) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:15]: what is this check for?  I guess you can delete it

**Review by BelehovEgor — APPROVED** [review]
check all cheking of lenght which looks like that i commented

File: src/UserService.Validation/User/CreateUserRequestValidator.cs (hunk 2 above)
Comment (MariyaSin) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 1 above)
Comment (MariyaSin) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 3]
```diff
@@ -2,50 +2,57 @@
 using LT.DigitalOffice.UserService.Models.Dto;
 using LT.DigitalOffice.UserService.Validation.User.Interfaces;
 using System.Linq;
+using System.Text.RegularExpressions;
 
 namespace LT.DigitalOffice.UserService.Validation.User
 {
     public class CreateUserRequestValidator : AbstractValidator<CreateUserRequest>, ICreateUserRequestValidator
     {
+        private static Regex NameRegex = new(@"\d");
```
Comment (TIove) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:11]: Mb better use “/D” and on 16 line for example don’t use negative operator ‘!’
It seems like double negative form

File: src/UserService.Validation/User/CreateUserRequestValidator.cs [hunk 4]
```diff
@@ -2,50 +2,57 @@
...
         public CreateUserRequestValidator()
         {
             RuleFor(user => user.FirstName)
                 .NotEmpty()
-                .MaximumLength(32).WithMessage("First name is too long.")
-                .MinimumLength(1).WithMessage("First name is too short.")
-                .Matches("^[A-Z][a-z]+$|^[А-ЯЁ][а-яё]+$").WithMessage("First name with error.");
+                .Must(x => !NameRegex.IsMatch(x))
+                .WithMessage("First name has incorrect format")
```
Comment (TIove) [inline src/UserService.Validation/User/CreateUserRequestValidator.cs:17]: Mb u should write it more specifically like “First name mustn’t have numbers”

**Review by TIove — APPROVED** [review]

File: src/UserService.Validation/User/CreateUserRequestValidator.cs (hunk 3 above)
Comment (MariyaSin) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs:11]: it is did not work((

File: src/UserService.Validation/User/CreateUserRequestValidator.cs (hunk 4 above)
Comment (MariyaSin) [reply src/UserService.Validation/User/CreateUserRequestValidator.cs]: done

**Review by djino97 — APPROVED** [review]

**Pull request merged**